src/auto-evo/steps/ModifyExistingSpecies.cs

```diff
@@ -199,45 +189,13 @@ public bool RunStep(RunResults results)
                 // Add these mutant species into a new miche to test them
                 foreach (var mutation in mutationsToTry)
                 {
-                    micheForMutationTests!.InsertSpecies(mutation.Item2, patch, null, cache, false, workMemory);
+                    miche.InsertSpecies(mutation.Item2, patch, null, cache, false, workMemory);
```

Comment:
Doesn't this modify the miche state now? Isn't this like super bad? I did test in game and things didn't crash with an exception but I wouldn't expect this to work at all...

Reply:
It does, but nothing outside population actually calls miche after that so it does not lead to any issues.

Reply:
But the mutations then *do* affect the population number simulation, which is different from before and makes the auto-evo prediction much less accurate.

I think I'd prefer to just have the deepcopy back and later consider something like implementing a separate miche type that has cached child nodes and it could be reinitialized to restore back to the default state. I think that would be easier than implementing like a overlay feature that allowed temporary species to be applied on top of the miche tree without affecting the underlying tree.

Reply:
Ah, also I realized one other thing is that the reason why this doesn't blow up is that each of these steps is handling just one patch meaning they aren't messing with any miche that another step would be using at the same time.